crates/testing/src/test_runner.rs

```diff
@@ -311,7 +311,7 @@ where
                         }
                     }
                     Err(e) => {
-                        panic!("Error Joining the test task {:?}", e);
```

Comment:
I'm not that familiar with async rust, but doesn't this change the semantics?

My understanding is: previously, the main thread would panic if one of the futures panics. But removing this means the main thread will ignore a thread panic and just log to stdout, which doesn't cause a test failure.

(I might be misunderstanding something of course.)

Reply:
It does change the semantics.  The futures are the testing tasks, and I think the issue I was seeing was with cancellation and only on one executor.  I can probably change it back now after after a few fixes